protocol/xmrtaker/instance.go

```diff
@@ -41,70 +32,28 @@ type Instance struct {
 
 // Config contains the configuration values for a new XMRTaker instance.
 type Config struct {
-	Backend                                backend.Backend
-	DataDir                                string
-	MoneroWalletFile, MoneroWalletPassword string
-	TransferBack                           bool
-	ExternalSender                         bool
+	Backend        backend.Backend
+	DataDir        string
+	TransferBack   bool
+	ExternalSender bool
 }
 
 // NewInstance returns a new instance of XMRTaker.
 // It accepts an endpoint to a monero-wallet-rpc instance where XMRTaker will generate
 // the account in which the XMR will be deposited.
 func NewInstance(cfg *Config) (*Instance, error) {
-	var (
-		address mcrypto.Address
-		err     error
-	)
-
 	// if this is set, it transfers all xmr received during swaps back to the given wallet.
 	if cfg.TransferBack {
-		address, err = getAddress(cfg.Backend, cfg.MoneroWalletFile, cfg.MoneroWalletPassword)
-		if err != nil {
-			return nil, err
-		}
-		cfg.Backend.SetBaseXMRDepositAddress(address)
+		cfg.Backend.SetBaseXMRDepositAddress(cfg.Backend.PrimaryWalletAddress())
```

Comment:
The `getAddress` function was removed in favor of asking the wallet what its primary address is. We cache the address when starting the wallet.